worker/exec/konnector.go

```diff
@@ -341,10 +346,14 @@ func (w *konnectorWorker) ensureFolderToSave(ctx *job.TaskContext, inst *instanc
 		for _, row := range res.Rows {
 			dir := &vfs.DirDoc{}
 			if err := couchdb.GetDoc(inst, consts.Files, row.ID, dir); err == nil {
-				if !strings.HasPrefix(dir.Fullpath, vfs.TrashDirName) {
-					count++
-					dirID = row.ID
+				if strings.HasPrefix(dir.Fullpath, vfs.TrashDirName) {
+					continue
+				}
+				if !hasCompatibleSourceAccountIdentifier(dir, sourceAccountIdentifier) {
```

Comment:
Since we require the folder to have a referenced `sourceAccountIdentifier` now, we could move the [early return when `acc` is `nil`](https://github.com/cozy/cozy-stack/pull/4360/files#diff-4719b9e6012f031281e0cc6d5914f4771b18c611845a260472d2024546835f24R365-R368) towards the top of the function.

Reply:
We don't require the folder to have a referenced `sourceAccountIdentifier`, it's optional.

Reply:
Ah yes, I missed the part where `hasCompatibleSourceAccountIdentifier` returns `true` if the folder does not have a `sourceAccountIdentifier`.
This answers my question about the data migration too.